code/obj/nuclearreactor/nuclearreactor.dm

```diff
@@ -839,6 +839,20 @@
 		return TRUE
 
 	tick(var/obj/projectile/P)
+		var/turf/simulated/T = get_turf(P)
+		if (istype(T, /turf/space/fluid)) //can't really calculate this, just make it do something at least
+			P.power -= 10
```

Comment:
The reagent generated is stored in `global.ocean_reagent_id` if you want to use that.